prefect_email/credentials.py

```diff
@@ -139,6 +141,7 @@ def example_get_server_flow():
             elif smtp_type == SMTPType.STARTTLS:
                 server = SMTP(smtp_server, smtp_port)
                 server.starttls(context=context)
-            server.login(self.username, self.password.get_secret_value())
+            if self.username is not None:
+                server.login(self.username, self.password.get_secret_value())
```

Comment:
ok, the logic is the same... but I do not understand your comment about Optional only being used when None is the default - what about smtp_type, etc?

Reply:
You are correct; those are my mistakes and they [shouldn't be type annotated with `Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) 

"Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional"

Reply:
Well, I've never used pydantic much either, so I learned something here too. It's a simple fix so might as well fold it in.

Reply:
I think everything is correct now.